consensus/core/src/round_prober.rs

```diff
@@ -148,18 +146,41 @@ impl<C: NetworkClient> RoundProber<C> {
 
         let mut highest_received_rounds =
             vec![vec![0; self.context.committee.size()]; self.context.committee.size()];
+        let mut highest_accepted_rounds =
+            vec![vec![0; self.context.committee.size()]; self.context.committee.size()];
+
+        let blocks = self
+            .dag_state
+            .read()
+            .get_last_cached_block_per_authority(Round::MAX);
+        let local_highest_accepted_rounds = blocks
+            .into_iter()
+            .map(|block| block.round())
+            .collect::<Vec<_>>();
+        let last_proposed_round = local_highest_accepted_rounds[own_index];
+
+        // For our own index, the highest recieved & accepted round is our last
+        // accepted round or our last proposed round.
         highest_received_rounds[own_index] = self.core_thread_dispatcher.highest_received_rounds();
+        highest_accepted_rounds[own_index] = local_highest_accepted_rounds;
         highest_received_rounds[own_index][own_index] = last_proposed_round;
+        highest_accepted_rounds[own_index][own_index] = last_proposed_round;
+
         loop {
             tokio::select! {
                 result = requests.next() => {
-                    let Some((peer, result)) = result else {
-                        break;
-                    };
+                    let Some((peer, result)) = result else { break };
                     match result {
-                        Ok(Ok(rounds)) => {
-                            if rounds.len() == self.context.committee.size() {
-                                highest_received_rounds[peer] = rounds;
+                        Ok(Ok((received, accepted))) => {
+                            if received.len() == self.context.committee.size()
+                                && (!self
+                                    .context
+                                    .protocol_config
+                                    .consensus_round_prober_probe_accepted_rounds()
+                                    || accepted.len() == self.context.committee.size())
```

Comment:
This condition seems not quite right. What if `consensus_round_prober_probe_accepted_rounds` is false and peers are returning empty `accepted`? Then `highest_accepted_rounds` for the `peer` will be set to empty and `compute_quorum_round()` will crash. It seems better to use separate if blocks for received and accepted.

Reply:
Great catch!! Thanks 🙇🏽